src/kktsystem.jl

```diff
@@ -313,3 +392,145 @@ function initialize!(kkt::SparseUnreducedKKTSystem)
 end
 
 is_reduced(::SparseUnreducedKKTSystem) = false
+
+#=
+    DenseKKTSystem
+=#
+
+struct DenseKKTSystem{T, VT, MT} <: AbstractKKTSystem{T, MT}
+    hess::MT
+    jac::MT
+    pr_diag::VT
+    du_diag::VT
+    diag_hess::VT
```

Comment:
In sparse KKT, `pr_diag` and `du_diag` are defined implicitly in the KKT's COO matrix, by adding artificial duplicate entries to the sparsity pattern. That allows to do the regularization seamlessly. 

In dense KKT, things are a bit different, as we cannot store `pr_diag` and `du_diag` directly in the dense KKT matrix. Each time we update the regularization, we need to update the diagonal terms in the dense KKT matrix, as 
```julia
for i in 1:n
    aug_com[i, i] = hess[i, i] + pr_diag[i]
```
In case `aug_com === hess`, we are adding two time the terms `pr_diag` in the second regularization. That's why I thought it would be better to store the diagonal terms of the Hessian matrix once, and then do the regularization as
```julia
for i in 1:n
    aug_com[i, i] = diag_hess[i] + pr_diag[i]
```
But there might be a better solution!